src/components/UserProfilePopUp/index.js

```diff
@@ -0,0 +1,23 @@
+import React, { useRef } from "react";
+import { withStyles } from "@material-ui/styles";
+
+import UserProfilePopUpHeader from "./UserProfilePopUpHeader";
+import UserMenu from "./UserMenu";
+import { useStyles } from "./styles";
+import useOutsideClick from "../Hooks/useOutsideClick";
+
+const UserProfilePopUp = ({ classes, handleClick }) => {
+  const wrapperRef = useRef(null);
+  useOutsideClick(wrapperRef, handleClick);
+
+  return (
+    <div ref={wrapperRef}>
+      <div className={classes.UserProfilePopUpContainer}>
+        <UserProfilePopUpHeader userName="waythingswork" remainingCredits="120" usedCredits="30" />
```

Comment:
Should the `userName` be coming from props?

Reply:
[done](https://github.com/singnet/refactored-dapp/pull/17/commits/dd3fd652b56d67e3282cc5c9dae05b198c426a3e)